internal/metrics/logfilemetricexporter/exporter.go

```diff
@@ -0,0 +1,119 @@
+package logfilemetricexporter
+
+import (
+	"strings"
+
+	"github.com/openshift/cluster-logging-operator/internal/tls"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	loggingv1a1 "github.com/openshift/cluster-logging-operator/apis/logging/v1alpha1"
+	"github.com/openshift/cluster-logging-operator/internal/collector"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	coreFactory "github.com/openshift/cluster-logging-operator/internal/factory"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+)
+
+const (
+	clusterLoggingPriorityClassName = "system-node-critical"
+	ExporterPort                    = int32(2112)
+	ExporterPortName                = "exporter-port"
+	exporterMetricsVolumeName       = "lfme-metrics"
+	ExporterMetricsSecretName       = "lfme-secret"
+	logContainers                   = "varlogcontainers"
+	logContainersValue              = "/var/log/containers"
+	logPods                         = "varlogpods"
+	logPodsValue                    = "/var/log/pods"
+	metricsVolumePath               = "/etc/logfilemetricexporter/metrics"
+)
+
+var (
+	defaultTolerations = []v1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		},
+		{
+			Key:      "node.kubernetes.io/disk-pressure",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		},
+	}
+)
+
+// ExporterResourceRequirements returns the resource requirements for a given metric-exporter implementation
+// or it's default if none are specified
+func ExporterResourceRequirements(exporter loggingv1a1.LogFileMetricExporter) v1.ResourceRequirements {
```

Comment:
Resources since Exporter is redundant with the package name

Reply:
Done